### PR TITLE
chore: use `thiserror` within crates

### PR DESCRIPTION
**Pull request by brunopgalvao**
`thiserror` is good for libraries:
- https://google.github.io/comprehensive-rust/error-handling/thiserror-and-anyhow.html

closes: https://github.com/r0gue-io/pop-cli/issues/8

### CODE REVIEW

File: crates/pop-contracts/src/new.rs [hunk 1]
```diff
@@ -1,30 +1,51 @@
...
-/// Create a new smart contract at `target`
-pub fn create_smart_contract(name: String, target: &Path) -> anyhow::Result<()> {
-	// In this code, out_dir will automatically join `name` to `target`,
-	// which is created prior to the call to this function
-	// So we must pass `target.parent()`
-	new_contract_project(&name, target.canonicalize()?.parent())
+#[derive(Error, Debug)]
+pub enum Error {
+	#[error("Failed to create new contract project: {0}")]
+	NewContractError(String),
```
Comment (weezy20) [inline crates/pop-contracts/src/new.rs:8]: I suggest to omit the `Error` suffix. When you're matching onn the error:
```rs
match fallible_operation() {
  Ok(_) => {}
  Err(e) => match e {
    Error::New,
    Error::Io,
    }
 ```
 helps eliminate some of the redundant visual noise.

File: crates/pop-contracts/src/utils/helpers.rs [hunk 2]
```diff
@@ -1,33 +1,40 @@
...
 
-// If the user specifies a path (which is not the current directory), it will have to manually
-// add a Cargo.toml file. If not provided, pop-cli will ask the user for a specific path. or ask
-// to the user the specific path (Like cargo-contract does)
-pub fn get_manifest_path(path: &Option<PathBuf>) -> anyhow::Result<ManifestPath> {
-	if path.is_some() {
-		let full_path: PathBuf =
-			PathBuf::from(path.as_ref().unwrap().to_string_lossy().to_string() + "/Cargo.toml");
+#[derive(Error, Debug)]
+pub enum Error {
```
Comment (weezy20) [inline crates/pop-contracts/src/utils/helpers.rs:9]: Why do individual modules have their own definition of an `Error` enum? For one crate, the idiomatic design is to have a single `error.rs` file, and collate all errors into that one enum. You can use struct variants if you feel one variant of `Error` can have multiple applications but fall under the same module.

Comment (brunopgalvao) [reply crates/pop-contracts/src/utils/helpers.rs]: Done.

**Comment by AlexD10S** [comment]
Nice work! 
I see here the tests passes in the CI but for me locally the test `test_new_parachain_command_execute` is failing, can you double check is passing for you?

**Review by AlexD10S — CHANGES_REQUESTED** [review]

File: crates/pop-contracts/src/test.rs [hunk 3]
```diff
@@ -1,46 +1,65 @@
...
-	cmd("cargo", vec!["test"]).dir(path.clone().unwrap_or("./".into())).run()?;
+pub fn test_smart_contract(path: &Option<PathBuf>) -> Result<(), Error> {
+	// Execute `cargo test` command in the specified directory.
+	let result = cmd("cargo", vec!["test"])
+		.dir(path.clone().unwrap_or_else(|| PathBuf::from("./")))
+		.run()
+		.map_err(|e| Error::TestCommand(format!("Cargo test command failed: {}", e)))?;
 
-	Ok(())
+	if result.status.success() {
```
[the remark this answers is not in the document]
Comment (brunopgalvao) [reply crates/pop-contracts/src/test.rs]: Good point. Removed.

**Comment by brunopgalvao** [comment]
> Nice work! I see here the tests passes in the CI but for me locally the test `test_new_parachain_command_execute` is failing, can you double check is passing for you?

I just ran `cargo test --all-features` and all tests pass.

**Review by AlexD10S — APPROVED** [review]

**Pull request merged**